wordpress-mobile/WordPressKit-iOS
# Refactor to use wpcom/v2 plans endpoint

This PR refactors the Plans remote to read from the `wpcom/v2/plans/mobile` endpoint instead of the `rest/v1.2/sites/site/plans` endpoint and `rest/v1.2/plans/features` endpoint. Remote models are refactored accordingly.  

The new end contains three models:
- RemoteWpcomPlan: The plan description.  Wpcom is specified to distinguish from Jetpack in anticipation of future Jetpack plans support.
- RemotePlanFeature: Descriptions of the features belonging to each plan.  This is largely unchanged, but moved to a new location. 
- RemotePlanGroup: Anticipates that plans can belong to one or more groups of plans. Currently Personal and Business groups are supported.

Code exclusive to the v1.2 endpoints is removed but anything still used by the v1.3 endpoint is retained. 

To test:
Confirm there are no build errors.
Run the `PlanServiceRemoteTests` test suite and confirm everything passes. 

@jklausa could I trouble you for a review of this one?  Domains relies on a call to the v1.3 endpoint and you're probably best able to tell if these changes will break things.
